pkg/util/readiness.go

```diff
@@ -45,24 +43,71 @@ func CheckCSVPhase(c client.Client, namespace string, csvNames ...string) health
 			return nil
 		}
 
-		for idx := range csvList.Items {
-			csv := &csvList.Items[idx]
-			_, exists := csvMap[csv.Name]
-			if exists {
-				if csv.Status.Phase != opv1a1.CSVPhaseSucceeded {
-					return fmt.Errorf("CSV %s is not in Succeeded phase", csv.Name)
-				} else if csv.Status.Phase == opv1a1.CSVPhaseSucceeded {
-					delete(csvMap, csv.Name)
-				}
-			}
+		return ValidateCSVsSucceeded(csvList, csvNames...)
+	}
+}
+
+func EnsureCSVsSucceeded(ctx context.Context, cli client.Client, namespace string, csvsToBeSuceeded ...string) error {
+
+	csvList, err := GetNamespaceCSVs(ctx, cli, namespace)
+	if err != nil {
+		return err
+	}
+
+	return ValidateCSVsSucceeded(csvList, csvsToBeSuceeded...)
+}
+
+func EnsureCSVsPresent(ctx context.Context, cli client.Client, namespace string, csvsToBePresent ...string) error {
+
+	csvList, err := GetNamespaceCSVs(ctx, cli, namespace)
+	if err != nil {
+		return err
+	}
+
+	return ValidateCSVsPresent(csvList, csvsToBePresent...)
+}
+
+func ValidateCSVsSucceeded(csvList *opv1a1.ClusterServiceVersionList, csvsToBeSucceeded ...string) error {
+
+	for _, csvName := range csvsToBeSucceeded {
+		csv, found := GetCSVByName(csvList, csvName)
+		if !found {
+			return fmt.Errorf("CSV %q not found in the list", csvName)
+		}
+
+		if csv.Status.Phase != opv1a1.CSVPhaseSucceeded {
+			return fmt.Errorf("CSV %q is not in the Succeeded phase; current phase: %s", csv.Name, csv.Status.Phase)
 		}
-		if len(csvMap) != 0 {
-			for csvName := range csvMap {
-				return fmt.Errorf("CSV %s is not found", csvName)
-			}
+	}
+	return nil
+}
+
+func ValidateCSVsPresent(csvList *opv1a1.ClusterServiceVersionList, csvsToBePresent ...string) error {
+
+	for _, csvName := range csvsToBePresent {
+		_, found := GetCSVByName(csvList, csvName)
+		if !found {
+			return fmt.Errorf("CSV %q not found in the list", csvName)
 		}
-		return nil
 	}
+	return nil
+}
+
+func GetCSVByName(csvList *opv1a1.ClusterServiceVersionList, name string) (*opv1a1.ClusterServiceVersion, bool) {
+
+	for _, csv := range csvList.Items {
+		if csv.Name == name {
+			return &csv, true
+		}
+	}
```

Comment:
pls query by index, here there is a creation of new var and it creates many not required values escaping the function scope.

Reply:
Done